### PR TITLE
Forcibly exit the JVM on older Clojure versions

### PR DESCRIPTION
**Pull request by vemv**
See https://github.com/nrepl/nrepl/issues/308

### CODE REVIEW

File: src/cider/nrepl.clj [hunk 1]
```diff
@@ -22,6 +22,11 @@
    [orchard.namespace]
    [orchard.java]))
 
+(when (and (< (-> *clojure-version* :major long) 2)
```
Comment (bbatsov) [inline src/cider/nrepl.clj:25]: I'd probably move the required version to a constant (a map would be best I guess), so it's clearer what we're doing.

File: src/cider/nrepl.clj [hunk 2]
```diff
@@ -22,6 +22,11 @@
...
+           (< (-> *clojure-version* :minor long) 9))
+  (.println System/err "cider-nrepl cannot be run with older Clojure versions. Exiting.")
```
Comment (bbatsov) [inline src/cider/nrepl.clj:27]: Might be good to mention the required Clojure version here.

**Comment by vemv** [comment]
Good suggestions, thanks!

File: src/cider/nrepl.clj [hunk 3]
```diff
@@ -22,10 +22,24 @@
...
+
+(when (or (< (-> *clojure-version* :major long)
+             (-> min-clojure-verion :major long))
+
+          (and (= (-> *clojure-version* :major long)
+                  (-> min-clojure-verion :major long))
+               (< (-> *clojure-version* :minor long)
+                  (-> min-clojure-verion :minor long))))
+  (try
+    (.println System/err (format "cider-nrepl cannot be run with older Clojure versions (found: %s). Exiting."
```
Comment (bbatsov) [inline src/cider/nrepl.clj:39]: I still think it'd be good to mention here the required version as well, so it's clearer for the users what they need.

Comment (vemv) [reply src/cider/nrepl.clj:39]: Sure thing! https://github.com/clojure-emacs/cider-nrepl/commit/4497c166919d84fa6a6833f2baaf35cfe0687d14

**Pull request merged**